src/Data/Generic/Rep/Enum.purs

```diff
@@ -0,0 +1,41 @@
+module Data.Generic.Rep.Enum
+  ( class GenericEnum
+  , genericSucc
+  , genericPred
+  , succ'
+  , pred'
+  ) where
+
+import Data.Generic.Rep
+import Data.Maybe (Maybe(..))
+import Data.Generic.Rep.Eq (class GenericEq, genericEq')
+import Data.Generic.Rep.Bounded (class GenericBottom, genericBottom',
+                                 class GenericTop, genericTop')
+import Prelude ((<$>))
+
+class (GenericEq a, GenericBottom a, GenericTop a) <= GenericEnum a where
```

Comment:
`GenericBottom` and `GenericTop` constraints should be moved to the `genericEnumSum` instance, I think.